Makefile

```diff
@@ -40,121 +40,88 @@ ifndef SR_XML
 #source
-	cp csharp/src/*.cs _build/csharp/autogen/src
-	mv _build/csharp/autogen/src/AssemblyInfo.cs _build/csharp/autogen/src/Properties/AssemblyInfo.cs
-	sed -i -e 's/1\.0\.0\.0/$(SDK_VERSION).0/g' _build/csharp/autogen/src/Properties/AssemblyInfo.cs
+	cp csharp/src/*.cs _build/default/csharp/autogen/src
+	mv _build/default/csharp/autogen/src/AssemblyInfo.cs _build/default/csharp/autogen/src/Properties/AssemblyInfo.cs
+	sed -i -e 's/1\.0\.0\.0/$(SDK_VERSION).0/g' _build/default/csharp/autogen/src/Properties/AssemblyInfo.cs
 #samples
-	cp -r csharp/samples/* _build/csharp/autogen/samples
+	cp -r csharp/samples/* _build/default/csharp/autogen/samples
 #other
-	sed -e 's/@SDK_VERSION@/$(SDK_VERSION)/g' csharp/README.dist > _build/csharp/autogen/README.txt
-	cp LICENSE _build/csharp/autogen/LICENSE.txt
-	sh windows-line-endings.sh _build/csharp/autogen
-
-java: main.native
-	mkdir -p _build/java/autogen/com/xensource/xenapi
-	mkdir -p _build/java/autogen/samples
-	./main.native -d _build/java/autogen -t java/templates
+	sed -e 's/@SDK_VERSION@/$(SDK_VERSION)/g' csharp/README.dist > _build/default/csharp/autogen/README.txt
+	cp LICENSE _build/default/csharp/autogen/LICENSE.txt
+	sh windows-line-endings.sh _build/default/csharp/autogen
+
+java: build
+	mkdir -p _build/default/java/autogen/com/xensource/xenapi
+	mkdir -p _build/default/java/autogen/samples
+	_build/default/java/main.exe -d _build/default/java/autogen -t java/templates
 #source
-	cp java/lib/com/xensource/xenapi/*.java _build/java/autogen/com/xensource/xenapi
-	sed -e 's/@SDK_VERSION@/$(SDK_VERSION)/g' java/lib/com/xensource/xenapi/Connection.java > _build/java/autogen/com/xensource/xenapi/Connection.java
+	cp java/lib/com/xensource/xenapi/*.java _build/default/java/autogen/com/xensource/xenapi
+	sed -e 's/@SDK_VERSION@/$(SDK_VERSION)/g' java/lib/com/xensource/xenapi/Connection.java > _build/default/java/autogen/com/xensource/xenapi/Connection.java
 #samples
-	cp java/samples/*.java _build/java/autogen/samples
+	cp java/samples/*.java _build/default/java/autogen/samples
 #other
-	sed -e 's/@SDK_VERSION@/$(SDK_VERSION)/g' java/lib/Makefile.dist > _build/java/autogen/Makefile
-	sed -e 's/@SDK_VERSION@/$(SDK_VERSION)/g' java/README.dist > _build/java/autogen/README.txt
-	cp LICENSE _build/java/autogen/LICENSE.txt
-	cp java/LICENSE.Apache-2.0.txt _build/java/autogen
-
-powershell: gen_powershell_binding.native
-	mkdir -p _build/powershell/autogen/src
-	mkdir -p _build/powershell/autogen/samples
-	./gen_powershell_binding.native -d _build/powershell/autogen
+	sed -e 's/@SDK_VERSION@/$(SDK_VERSION)/g' java/lib/Makefile.dist > _build/default/java/autogen/Makefile
+	sed -e 's/@SDK_VERSION@/$(SDK_VERSION)/g' java/README.dist > _build/default/java/autogen/README.txt
+	cp LICENSE _build/default/java/autogen/LICENSE.txt
+	cp java/LICENSE.Apache-2.0.txt _build/default/java/autogen
+
+powershell: build
```

Comment:
These targets could be made into `jbuilder` targets and moved to the various `jbuild` files, however in this case I think it is more readable if we leave them in the `Makefile`